Mathlib/Algebra/Group/Opposite.lean

```diff
@@ -102,11 +102,19 @@ instance leftCancelSemigroup [RightCancelSemigroup α] : LeftCancelSemigroup α
   { MulOpposite.semigroup α with
     mul_left_cancel := fun _ _ _ H => unop_injective <| mul_right_cancel <| op_injective H }
```

Comment:
If you swap the order of the declarations, you should be able to prove this one in terms of the new one.

Reply:
Your suggestion worked.  Several of the `MulOpposite.[something] α with` could be removed and Lean would not complain.  Is there a reason to keep them?  I have been following as a spectator the "`let` in instances" progress, but I do not know if this is part of it.